# **Income Prediction**

In [65]:
import pandas as pd

In [69]:
data = pd.read_csv('Income Prediction.csv')
data.head()

age        wc   education  marital status    race   gender  hours per week  \
0   38   Private     HS-grad        Divorced   White     Male              40   
1   28   Private   Bachelors         Married   Black   Female              40   
2   37   Private     Masters         Married   White   Female              40   
3   31   Private     Masters   Never-married   White   Female              50   
4   42   Private   Bachelors         Married   White     Male              40   

  IncomeClass  
0       <=50K  
1       <=50K  
2       <=50K  
3        >50K  
4        >50K

In [71]:
data.isnull().sum(axis=0)

age               0
wc                0
education         0
marital status    0
race              0
gender            0
hours per week    0
IncomeClass       0
dtype: int64

In [73]:
data.dtypes

age                int64
wc                object
education         object
marital status    object
race              object
gender            object
hours per week     int64
IncomeClass       object
dtype: object

# Create Dummy Variables

In [76]:
data_prep= pd.get_dummies(data, drop_first=True)

In [78]:
X= data_prep.iloc[:,:-1]
Y= data_prep.iloc[:,-1]

In [80]:
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test=\
    train_test_split(X, Y, test_size = 0.3, random_state = 1234, stratify=Y)

In [82]:
from sklearn.tree import DecisionTreeClassifier

In [84]:
dtc= DecisionTreeClassifier(random_state=1234)
dtc.fit(X_train, Y_train)
Y_predict= dtc.predict(X_test)

# Confusion Matrix

In [87]:
from sklearn.metrics import confusion_matrix
cm= confusion_matrix(Y_test, Y_predict)
print(cm)

[[3814  559]
 [ 800  764]]


In [91]:
score = dtc.score(X_test, Y_test)
print(score)

0.7710965133906014


# Conclusion

<font size="4px"><p>True negatives: 3814, True positives:  764, False positives: 559, False negatives: 800</p>
<font size="4px"><p>The Accuracy is 77%, only 23% of false perdiction</p>